test/integration/utils.test.js

```diff
@@ -189,4 +189,20 @@ describe(Support.getTestDialectTeaser('Utils'), function() {
       expect(Utils.singularize('status')).to.equal('status');
     });
   });
+
+  describe('formatReferences', function () {
+    ([
+      [{referencesKey: 1}, {references: {model: undefined, key: 1}}],
+      [{references: 'a'}, {references: {model: 'a', key: undefined}}],
+      [{references: 'a', referencesKey: 1}, {references: {model: 'a', key: 1}}],
+      [{references: {model: 1}}, {references: {model: 1}}]
+    ]).forEach(function (test) {
+      var input  = test[0];
+      var output = test[1];
+
+      it('converts ' + JSON.stringify(input) + ' to ' + JSON.stringify(output), function () {
+        expect(Utils.formatReferences(input)).to.deep.equal(output);
+      });
+    });
+  });
```

Comment:
This is actually a unit test, not a integration test. The test itself though is pretty useless right now though. You test your helper function but not if this would work in real code, since you would want to test if your helper function was actually called by the code and everything worked as intended. So a better way to test this would be to write unit tests to test the generated sql for each statement that may use a references object.